src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle= 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Blue = 'blue',
+  Green = 'green',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || b <= 0) {
+      throw new Error('The side length cannot be less than or equal to 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('The side length cannot be greater'
+      + 'than or equal to sum of the others');
+    }
+
+    this.shape = Shape.Triangle;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
You don`t need it when you added public or private in constructor arguments

Reply:
agree, fix it everywhere please